### PR TITLE
CLN: Minor cleanup of deprecations in parser code

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

Some spaghetti that I noticed and was annoying.

### CODE REVIEW

File: pandas/io/parsers/readers.py [hunk 1]
```diff
@@ -1045,8 +1042,8 @@ def _clean_options(self, options, engine):
 
         validate_header_arg(options["header"])
 
-        for arg in _deprecated_args:
-            parser_default = _c_parser_defaults[arg]
+        for arg in _deprecated_defaults.keys():
```
Comment (jbrockmendel) [inline pandas/io/parsers/readers.py:1045]: NBD but i think ".keys()" is unnecessary here

Comment (lithomas1) [reply pandas/io/parsers/readers.py]: Thanks for the catch.

File: pandas/io/parsers/readers.py [hunk 2]
```diff
@@ -442,7 +440,6 @@
 }
 
 _deprecated_defaults: dict[str, Any] = {"error_bad_lines": None, "warn_bad_lines": None}
-_deprecated_args: set[str] = {"error_bad_lines", "warn_bad_lines"}
```
Comment (jbrockmendel) [inline pandas/io/parsers/readers.py:445]: looks like both of these are present in the parsers.pyi file but not in parsers.pyx.  Does one of those need to be updated?

**Comment by jreback** [comment]
thanks @lithomas1

**Pull request merged**